### ISSUE TITLE
JavaScript Course, More Testing Lesson: Include a video on dependency injection in the assignments

### ISSUE THREAD

**Issue by CCeph** [TheOdinProject/curriculum#26391]
### Describe your suggestion

The first assignment for this lesson asks people to watch the "mocking" videos from this fun fun function [series](https://www.youtube.com/watch?v=3PjdxjWK0F0&ab_channel=FunFunFunction). Dependency injection is used extensively in this series, but up until this point in TOP, it is never taught. While there is brief mention in the series that fun fun function has a separate video on the basics of dependency injection, some might consider it an unnecessary detour from the curriculum.

Hence, I would like this [video](https://www.youtube.com/watch?v=0X1Ns2NRfks&t=776s&ab_channel=FunFunFunction) to be added to the assignments.

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-more-testing

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

cceph

**Comment by codyloyd** [comment]
I think this might be worth discussing a bit.  Dependency injection is an important topic and should probably be covered _somewhere_

HOWEVER: I am not sure that assigning another 20minute video here is the best plan.  It might be worth finding a shorter article or more straightforward video that simply defines the concept, and then referring to it when assigning the mocking vids.

something like this:

> If you haven’t already, watch the ‘mocking’ videos from [this series](https://www.youtube.com/watch?v=3PjdxjWK0F0). He refers to the concept of 'dependency injection' quite a bit in this video, so if you're unfamiliar with that concept you might want to read this INSERT-ARTICLE-OR-VIDEO-HERE first.

**Comment by ManonLef** [comment]
I agree with @codyloyd on this. The testing lesson already has an extreme amount of video content, especially since the fun fun functions video series is so lengthy. 

I have a few articles on my own list for this that could be a fit:
- [short read, explains DI](https://javascript.plainenglish.io/dependency-injection-in-javascript-539f7f6cfbee)
- [this article has a good explanation](https://www.meticulous.ai/blog/mocking-a-javascript-class-with-jest-two-ways-to-make-it-easier) but we might have to limit the reader to the section that is relevant. 
- [more in depth article and perhaps a bit opinionated](https://blog.bitsrc.io/dependency-injection-in-javascript-the-best-tool-youre-not-using-for-your-tests-4e83aca7579f)

**Comment by ManonLef** [comment]
@CCeph What do you think about the above?

**Comment by CCeph** [comment]
I read all three articles, and I think the first one is best. It's simple and straight to the point.

I found the second article's dependency injection section to be quite complicated with a lot of jargon. It makes dependency injection look like a complicated scary thing when it is not. The third article is a bit longer, very opinionated, with examples that are a tad more complex.

I also still think that the fun fun function dependency injection can be a good resource if students are told to only watch until 3:50. In 4 minutes, the video explains what dependency injection and uses the same dependency (fetch) that's injected in the mocking series. The student would only be missing the why of using dependency injection, which can be explained in a paragraph.

All in all, I think simpler is better here. In my experience, mocking alone can be a mind bender, so it'd be good if the dependency injection resource is as simple as possible. I'd be happy to either include the first article, or the first 4 minutes of the fun fun function video. What do you think?

**Comment by CouchofTomato** [comment]
@codyloyd @ManonLef 

Can we get an opinion on the last comment please.

**Comment by ManonLef** [comment]
@CCeph I am sorry this took a while, I've been mostly absent in the past months due to a new programming adventure :). 

I still favour the article but I'd be okay with including both as well with the video clearly noted as only until the specified time.
Since it's been a while, would you still be interested in being assigned this?

**Comment by CCeph** [comment]
Sounds good! I will look into it hopefully over the next week.

**Comment by CCeph** [comment]
I created the pull request for it. It is my first one, so I hope it follows all the rules. If it does not, please let me know and I'll update it right away.
